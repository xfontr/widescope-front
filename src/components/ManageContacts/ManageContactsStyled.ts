import styled from "styled-components";

const ManageContactsStyled = styled.div`
  position: fixed;
  z-index: 9999999;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.6);

  display: flex;
  justify-content: center;
  align-items: center;

  .contacts {
    border-radius: ${({ theme }) => theme.shapes.radiusSmall};
    padding: calc(${({ theme }) => theme.spacing.paddingBig} * 2);
    background-color: ${({ theme }) => theme.colors.primary};
    min-height: 40vh;
    min-width: 40vw;

    &__title {
      margin: 0;
      margin-bottom: ${({ theme }) => theme.spacing.gapSmall};
    }
  }
`;

export default ManageContactsStyled;
