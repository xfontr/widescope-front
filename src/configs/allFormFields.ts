import {
  FormComponents,
  FormFields,
} from "../components/RenderForm/RenderFormTypes";

export interface FormValue {
  type?: React.HTMLInputTypeAttribute;
  label: string;
  name: FormFields;
  belongsTo: FormComponents[];
  placeholder?: string;
  customClass?: string;
  customGroupClass?: string;
  renderAs?: "input" | "textarea";
}

export const allFormFields: Record<FormFields, FormValue> = {
  name: {
    type: "text",
    label: "Name",
    name: "name",
    belongsTo: ["logIn", "signUp"],
    placeholder: "John Doe",
    customClass: "form__input--user",
  },

  password: {
    type: "password",
    label: "Password",
    name: "password",
    belongsTo: ["logIn", "signUp"],
    customClass: "form__input--password",
  },

  repeatPassword: {
    type: "password",
    label: "Repeat password",
    name: "repeatPassword",
    belongsTo: ["signUp"],
  },

  email: {
    type: "email",
    label: "Email address",
    name: "email",
    belongsTo: ["signUp"],
  },

  projectName: {
    type: "text",
    label: "Name",
    name: "name",
    placeholder: "Music app",
    belongsTo: ["createProject", "updateProject"],
  },

  repository: {
    type: "text",
    label: "Repository URL",
    name: "repository",
    placeholder: "http://www.github.com/music-app",
    belongsTo: ["createProject", "updateProject"],
  },

  logo: {
    type: "file",
    label: "Project logo",
    name: "logo",
    belongsTo: ["createProject"],
    customGroupClass: "area",
  },

  logoUpdate: {
    type: "file",
    label: "Project logo",
    name: "logoUpdate",
    belongsTo: ["updateProject"],
    customGroupClass: "area",
  },

  technologyFront: {
    type: "text",
    label: "Frontend main library or framework",
    name: "technologyFront",
    placeholder: "React",
    belongsTo: ["createProject", "updateProject"],
  },

  technologyBack: {
    type: "text",
    label: "Backend main library or framework",
    name: "technologyBack",
    placeholder: "React",
    belongsTo: ["createProject", "updateProject"],
  },

  description: {
    label: "Description",
    name: "description",
    placeholder: "Music app is about sharing music between users",
    belongsTo: ["createProject", "updateProject"],
    renderAs: "textarea",
    customGroupClass: "area",
  },
};
