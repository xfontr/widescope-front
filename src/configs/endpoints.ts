const endpoints = {
  usersRoot: "/users",
  signUp: "/users/sign-up",
  logIn: "/users/log-in",

  projectsRoot: "/projects",
  getAll: "/projects/all",
  projectById: "/projects/",
  projectsByAuthor: "/projects/author/",
  createProject: "/projects/new",
};

export default endpoints;
