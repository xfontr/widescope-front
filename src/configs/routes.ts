const routes = {
  root: "/",
  home: "/home",
  signUp: "/sign-up",
  logIn: "/log-in",
  explore: "/explore",
  projectDetails: "/project/:projectId",
};

export default routes;
