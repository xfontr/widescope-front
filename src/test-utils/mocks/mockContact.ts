import IContact from "../../types/IContact";

const mockContact: IContact = {
  id: "contact id",
  name: "contact name",
  isFriend: false,
};

export default mockContact;
