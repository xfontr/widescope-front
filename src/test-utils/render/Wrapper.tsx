import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { store } from "../../app/store";
import GlobalStyles from "../../styles/GlobalStyles";
import styledMainTheme from "../../styles/styledMainTheme";

interface WrapperProps {
  children: JSX.Element | JSX.Element[];
}

const Wrapper = ({ children }: WrapperProps): JSX.Element => {
  return (
    <ThemeProvider theme={styledMainTheme}>
      <Provider store={store}>
        <BrowserRouter>
          <GlobalStyles />
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>
  );
};

export default Wrapper;
