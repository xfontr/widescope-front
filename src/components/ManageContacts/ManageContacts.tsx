import ManageContactsStyled from "./ManageContactsStyled";
import SearchBar from "../SearchBar/SearchBar";

interface ManageContactsProps {
  visibility: boolean;
}

const ManageContacts = ({ visibility }: ManageContactsProps) => {
  return (
    <>
      {visibility && (
        <ManageContactsStyled>
          <section className="contacts">
            <h2 className="contacts__title page__title">Your contacts</h2>
            <SearchBar />
          </section>
        </ManageContactsStyled>
      )}
      {!visibility && <></>}
    </>
  );
};

export default ManageContacts;
