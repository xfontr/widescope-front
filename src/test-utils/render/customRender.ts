import { render } from "@testing-library/react";
import { Wrapper } from "./Wrapper";

const customRender = (view: JSX.Element, ...options: any) =>
  render(view, { wrapper: Wrapper, ...options });

export * from "@testing-library/react";

export { customRender as render };
