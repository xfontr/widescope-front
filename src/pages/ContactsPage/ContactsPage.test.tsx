import { render as reactRender } from "@testing-library/react";
import { allFormFields } from "../../configs/allFormFields";
import mockContact from "../../test-utils/mocks/mockContact";
import { render, screen } from "../../test-utils/render/customRender";
import { WrapperWithMockStore } from "../../test-utils/render/Wrapper";
import ManageContacts from "./ContactsPage";

describe("Given a ManageContacts component", () => {
  describe("When instantiated", () => {
    test("Then there should be a heading and a search form", async () => {
      render(<ManageContacts />);

      const manageContacts = [
        screen.getByRole("heading", { name: "Meet other authors", level: 2 }),
        screen.getByLabelText(allFormFields.search.label),
        screen.getByRole("heading", { name: "Your friends", level: 3 }),
      ];

      manageContacts.forEach((element) => expect(element).toBeInTheDocument());
    });

    test("Then it should load all the friends at the store after fetching the database", async () => {
      reactRender(<ManageContacts />, { wrapper: WrapperWithMockStore });

      const friends = await screen.findAllByRole("heading", {
        name: mockContact.name,
        level: 3,
      });

      friends.forEach((friend) => expect(friend).toBeInTheDocument());
    });
  });

  describe("When instantiated but the user has no friends", () => {
    test("Then it should not load any friend", () => {
      render(<ManageContacts />);

      const friends = screen.queryAllByRole("heading", {
        name: mockContact.name,
        level: 3,
      });

      friends.forEach((friend) => expect(friend).not.toBeInTheDocument());
    });
  });
});
