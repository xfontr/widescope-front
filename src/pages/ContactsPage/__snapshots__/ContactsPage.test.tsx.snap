// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ManageContacts component When instantiated Then it should match the snapshot 1`] = `
<div
  className="sc-kDDrLX devrGj"
>
  <h2
    className="page__title"
  >
    Meet other 
    <span
      className="page__title--bold"
    >
      authors
    </span>
  </h2>
  <form
    className="sc-bczRLJ jzjTXG search-bar__container"
    onSubmit={[Function]}
  >
    <div
      className="sc-dkzDqf bRizeo area"
    >
      <label
        className="sc-hKMtZM chFrlm"
        htmlFor="search"
      >
        Find a friend
      </label>
      <input
        className="sc-eCYdqJ dwpIGp  search-bar"
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="Search..."
        type=""
        value=""
      />
    </div>
    <button
      className="sc-gKXOVf fgJUzp undefined button--default "
      onClick={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      Add friend
    </button>
  </form>
  <section
    className="friends"
  >
    <h3
      className="page__subheading"
    >
      Your friends
    </h3>
    <ul
      className="sc-jqUVSM hKyCQk"
    >
      <span>
        You don't seem to have any contacts :(
      </span>
    </ul>
  </section>
</div>
`;
